### DISCUSSION TITLE
IDE Help for ERRID enumeration

### DISCUSSION THREAD

**Discussion by AdamSpeight2008** [dotnet/roslyn#51363] [Ideas]
This is very specific to the Roslyn project, it would be cool if we could see the error message (from the resources file) for `ERRID` enumerations, on mouse over. That we can see what the arguments need to be filled in, as well see if that error is applicable to the situation required.

**Comment by CyrusNajmabadi** [comment]
We coudl do what the IDE does.  The enum constant must match teh english form of the string.  i.e.:

`Cannot_generate_code_for_unsupported_operator_0` is the name of the value: `"Cannot generate code for unsupported operator '{0}'"`

That way you can both just read things nicely, and also see what the placeholders correspond to.

**Reply by AdamSpeight2008** [reply]
XML Documentation on each member? I would be cool if it could also pull from the resx. g
```vbnet

        ''' <summary><include file="VBResources.resx" path="/data[@name='ERR_BlockLocalShadowing1']::value/" /></summary>
        ''' <code>Variable '{0}' hides a variable in an enclosing block.</code>
        ERR_BlockLocalShadowing1 = 30616

````
